lib/collection/proxy.js

```diff
@@ -0,0 +1,206 @@
+* @property {String=} [host] The proxy server host name @default would be '' [Empty String] represents no proxy
+* @property {String=} [port] The proxy server port @default would be '80'
+* @property {Boolean=} [tunnel] The option to provide whether tunneling needs to be done while proxying the request,
+* @default would be true.
+*
+* @example <caption>JSON definition of an example proxy object</caption>
+* {
+*     url: 'https://example.com',
+*     protocol: 'http',
+*     host: 'proxy.com',
+*     port: 88,
+*     tunnel: true
+* }
+*
+* Incase of ProxyString format
+* @param {String=} [options] supported formats return by the PAC files as provided here
+* (https://web.archive.org/web/20070602031929/http://wp.netscape.com/eng/mozilla/2.0/relnotes/demo/proxy-live.html)
+* @param {String=} [url] The url for which the proxy needs to be configured @default would '*' means select all url.
+
+* @example
+* var proxyObj = new Proxy('PROXY [https://proxy.com]:88', 'http://example.com')
+* {
+*     url: 'http://example.com',
+*     protocol: 'http',
+*     host: 'proxy.com',
+*     port: '88',
+*     tunnel: true
+* }
+*/
+_.inherit((
+    Proxy = function PostmanProxy (options, url) {
+        // this constructor is intended to inherit and as such the super constructor is required to be executed
+        Proxy.super_.call(this, options);
+
+        // Assign defaults before proceeding
+        _.assign(this, /** @lends Variable.prototype */ {
+            /**
+             * The url for which the proxy has been associated with.
+             * @type {String}
+             */
+            url: SELECTALL,
+
+            /**
+             * @type {String}
+             * @default 'http'
+             */
+            protocol: HTTP,
+
+            /**
+             * @type {String}
+             * @default ''
+             */
+            host: E,
+
+            /**
+             * @type {String}
+             * @default '80'
+             */
+            port: DEFAULT_PORT,
+
+            /**
+             * @type {Boolean}
+             * @default true
+             */
+            tunnel: true
+        });
+
+        if (!options) { return; } // in case definition object is missing, there is no point moving forward
+
+        /**
+        * If the options and url is a string then we need to parse the input and update it.
+        */
+        _.isString(options) && (options = Proxy.parse(options, url));
+
+        /**
+        * Make sure the 'url' key property is available in the options provided or generated
+        */
+        this.update(options);
+
+    }), Property);
+
+_.assign(Proxy.prototype, /** @lends Proxy.prototype */ {
+    update: function (options) {
+        if (!_.isObject(options)) {
+            return;
+        }
+        _.assign(this, {
+            /**
+             * The url for which the proxy has been associated with.
+             * @type {String}
+             */
+            url: options.url || SELECTALL,
```

Comment:
yeah. but as we support wild cards aswell, i don't think so `Url` is the best option for it.